api/src/Tests/Cases/CycleTest.php

```diff
@@ -40,25 +37,21 @@ protected function tearDown(): void
 
     protected function addCycle(): string
     {
-        $data = $this->runSuccessJsonRequest('GET', '/cycle', null, null, 200, null, '/cycle');
+        $data = testRun::testRun($this, 'GET', '/cycle')->run();
         $initial_ids = [];
         foreach ($data->data as $item) {
             $initial_ids[] = $item->id;
         }
 
         $when = date('Y-m-d', strtotime('-1 month'));
         $to = date('Y-m-d', strtotime('+1 month'));
-        $data = $this->runSuccessJsonRequest(
-            'POST',
-            '/cycle',
-            null,
-            ['date_from' => $when, 'date_to' => $to],
-            201,
-            null,
-            '/cycle'
-        );
-
-        $data2 = $this->runSuccessJsonRequest('GET', '/cycle/'.$data->id, null, null, 200, null, '/cycle/{id}');
+        $data = testRun::testRun($this, 'POST', '/cycle')
```

Comment:
Is the plan to follow up with this and break this out into nice segmented tests as well?

Reply:
We can continue to break things up and refine them yes.  Always in the mode to keep expanding and improving tests.